0000-catalog.md

```diff
@@ -0,0 +1,779 @@
+            string_value="30 days from sale"
+        ),
+        # ... Additional Product Properties
+    ])
+```
+
+A catalog product, from a data model perspective, is a grid product. With the 
+differentiators being:
+- It contains the additional fields defined by the catalog_product schema
+- It's stored in the catalog_product state location within the merkle tree
+
+## Transaction Payload and Execution
+
+The following payload contains an action enum and the associated action
+payload.This allows for the action payload to be dispatched to the appropriate 
+logic. Only the defined actions (enum) are available, and only one Action 
+should be performed in a payload.
+
+### CatalogPayload Transaction
+
+```
+message CatalogPayload {
+    enum Actions {
+        UNSET_ACTION = 0;
+        CATALOG_CREATE = 1;
+        CATALOG_UPDATE = 2;
+        CATALOG_DELETE = 3;
+        CATALOG_PRODUCT_CREATE = 100;
+        CATALOG_PRODUCT_UPDATE = 101;
+        CATALOG_PRODUCT_DELETE = 102;
+        CATALOG_PRODUCT_SET_STATUS = 103;
+    }
+
+    Action action = 1;
+
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    CatalogCreateAction catalog_create = 3;
+    CatalogUpdateAction catalog_update = 4;
+    CatalogDeleteAction catalog_delete = 5;
+    CatalogProductCreateAction catalog_product_create = 6;
+    CatalogProductUpdateAction catalog_product_update = 7;
+    CatalogProductDeleteAction catalog_product_delete = 8;
+    CatalogProductSetStatusAction set_catalog_product_status = 9;
+}
+```
+
+## Catalog Actions
+
+### CatalogCreateAction
+
+CatalogCreateAction adds a new catalog to state. The transaction should be
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the create transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+
+```
+message CatalogCreateAction {
+    // GS1 Company Prefix from owner + catalog_id are use as
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id already exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogCreateAction must include:
+
+Address of the Agent submitting the transaction
+Address of the Organization the Catalog is being created for
+Address of the Catalog to be created
+
+The outputs for CatalogCreateAction must include:
+
+Address of the Catalog created
+
+### CatalogUpdateAction
+
+CatalogUpdateAction adds a new catalog to state. The transaction should be
```

Comment:
Should this say updates an existing catalog in state.

Reply:
✅

Reply:
This still says "new catalog" should say existing

Reply:
fixed ✅